Add what-is-lightning-network to bitcoinology (#307)
bitcointranscripts/bitcointranscripts: bitcoinology/what-is-lightning-network.md

It's intended to be sort of halfway between serious academic stuff on the one hand and on the other hand just like culture and just enjoyment stuff. So this first, it's like a test meetup. We're just seeing how it will go. So we decided that a good topic would be the usage of Lightning, Lightning Network, as a sort of second layer of Bitcoin.

 And I'm gonna go through, what I'm gonna do in this talk is basically we're gonna have two half to this session. The first half is me giving a talk, Traditional presentation. You can interrupt with questions if you like, that's fine. If you do, I will have to repeat them, reminder to self. So I'm gonna explain some of the history behind Lightning Network, where it came from. Because if you only know a little bit about Bitcoin, and we just established with like a hands up, most people here know quite a bit about Bitcoin, but some people may be watching, or some people in this room right now maybe don't know that much about Bitcoin, and they maybe know even less about Lightning.
 

>   So I wanna give the context, where it comes from, why it's useful today, and hopefully that will lead into what it's really like to use the Lightning Network today as a payment system. And we're actually gonna do that in practice in the second hour. So let me kick off by first of all going to the start of this presentation. 

Reviewer: Inconsistent spacing here. Can we standardize how these paragraphs start?

> ## The next few years

Reviewer: This heading breaks up the paragraphs. Can we find a better break to put the header?

> ## Lightning Network paper

Reviewer: Same header issue here

> what we're gonna see in the next couple of years is in the background, something else was happening. So this is January 2016, roughly. We have these two guys, Taj Dreiger and Joseph Poon, writing a paper called The Lightning Network Paper. Initially it was something just in the back quarters because it was just technical people, a few developers here and there and so on, researchers who were even remotely interested in this. Because actually this paper was kind of strange. It was written in a kind of strange way. I don't know how many people... Hands up if you have actually read the Lightning Network paper. I mean, more than like a few lines. More than a few lines. A few lines doesn't count. All right? There's a few. Well, well done. You're obviously very masochistic people because that paper is not an easy read. And this illustrates, I think, perfectly. This die is one of the final diagrams. It just shows you how complicated the structure of what they were describing was. So it was very difficult to understand, and it was based around a couple of little hacks because actually Bitcoin's blockchain intrinsically didn't support it. Now before I run on, what on earth am I talking about, the Lightning Network paper? What is the Lightning Network, just for the people who don't know what it is? It's basically the idea that given everything I've said in the last 10 minutes, we need a system where we can transfer Bitcoin without actually putting it on the Bitcoin blockchain. Because the Bitcoin blockchain is intrinsically pretty expensive to use. Intrinsically, once enough people start using it. So their idea was, let's make transactions of Bitcoin back and forth, not just back and forth between two people, but between anyone in the world and me, and have it not actually touch the blockchain until some later time. And we'll go into that later, how that works. So that was their idea, but to actually make it work was this incredibly complicated Byzantine structure. And part of what made it complicated was something called malleability. So malleability is just a fancy word for change, really. If something's changeable, then it's malleable. And the problem is that intrinsically at that time, transactions on Bitcoin's blockchain were in some sense malleable in that when you created them, you could then create another version of them that was still valid before it got committed. So I know I'm getting a bit technical there, but basically there was a problem with the Bitcoin's blockchain that made it too changeable. And a solution came out called SegWit, but SegWit got all mixed up with something called the block size war. So while in 2016 these guys were coming up with this clever new scheme and some other people were getting excited about this clever new scheme, even though it was pretty complicated, they were still excited because we could get these transactions off the blockchain. A lot of other people were still arguing about, no, no, no, let's make the block sizes bigger because that way we can get more and more transactions on the blockchain. Okay, so the argument was about should we go off-chain or should we make the blockchain bigger to support more and more transactions. So Jonathan Beer here wrote a book recently called the Block Size War, I think about last year sometime. And it's quite interesting, if you don't know the history of the Block Size War, it is kind of interesting, but that's not the topic of our talk.

Reviewer: ```suggestion
what we're gonna see in the next couple of years is in the background, something else was happening. So this is January 2016, roughly. We have these two guys, Taj Dreiger and Joseph Poon, writing a paper called The Lightning Network Paper. Initially it was something just in the back quarters because it was just technical people, a few developers here and there and so on, researchers who were even remotely interested in this. Because actually this paper was kind of strange. It was written in a kind of strange way. I don't know how many people... Hands up if you have actually read the Lightning Network paper. I mean, more than like a few lines. More than a few lines. A few lines doesn't count. All right? There's a few. Well, well done. You're obviously very masochistic people because that paper is not an easy read. And this illustrates, I think, perfectly. This die is one of the final diagrams. It just shows you how complicated the structure of what they were describing was. So it was very difficult to understand, and it was based around a couple of little hacks because actually Bitcoin's blockchain intrinsically didn't support it. Now before I run on, what on earth am I talking about, the Lightning Network paper? What is the Lightning Network, just for the people who don't know what it is? It's basically the idea that given everything I've said in the last 10 minutes, we need a system where we can transfer Bitcoin without actually putting it on the Bitcoin blockchain. Because the Bitcoin blockchain is intrinsically pretty expensive to use. Intrinsically, once enough people start using it. So their idea was, let's make transactions of Bitcoin back and forth, not just back and forth between two people, but between anyone in the world and me, and have it not actually touch the blockchain until some later time. And we'll go into that later, how that works. So that was their idea, but to actually make it work was this incredibly complicated Byzantine structure. And part of what made it complicated was something called malleability. So malleability is just a fancy word for change, really. If something's changeable, then it's malleable. And the problem is that intrinsically at that time, transactions on Bitcoin's blockchain were in some sense malleable in that when you created them, you could then create another version of them that was still valid before it got committed. So I know I'm getting a bit technical there, but basically there was a problem with the Bitcoin's blockchain that made it too changeable. And a solution came out called SegWit, but SegWit got all mixed up with something called the block size war. So while in 2016 these guys were coming up with this clever new scheme and some other people were getting excited about this clever new scheme, even though it was pretty complicated, they were still excited because we could get these transactions off the blockchain. A lot of other people were still arguing about, no, no, no, let's make the block sizes bigger because that way we can get more and more transactions on the blockchain. Okay, so the argument was about should we go off-chain or should we make the blockchain bigger to support more and more transactions. So Jonathan Bier here wrote a book recently called the Block Size War, I think about last year sometime. And it's quite interesting, if you don't know the history of the Block Size War, it is kind of interesting, but that's not the topic of our talk.
```

> what we're gonna see in the next couple of years is in the background, something else was happening. So this is January 2016, roughly. We have these two guys, Taj Dreiger and Joseph Poon, writing a paper called The Lightning Network Paper. Initially it was something just in the back quarters because it was just technical people, a few developers here and there and so on, researchers who were even remotely interested in this. Because actually this paper was kind of strange. It was written in a kind of strange way. I don't know how many people... Hands up if you have actually read the Lightning Network paper. I mean, more than like a few lines. More than a few lines. A few lines doesn't count. All right? There's a few. Well, well done. You're obviously very masochistic people because that paper is not an easy read. And this illustrates, I think, perfectly. This die is one of the final diagrams. It just shows you how complicated the structure of what they were describing was. So it was very difficult to understand, and it was based around a couple of little hacks because actually Bitcoin's blockchain intrinsically didn't support it. Now before I run on, what on earth am I talking about, the Lightning Network paper? What is the Lightning Network, just for the people who don't know what it is? It's basically the idea that given everything I've said in the last 10 minutes, we need a system where we can transfer Bitcoin without actually putting it on the Bitcoin blockchain. Because the Bitcoin blockchain is intrinsically pretty expensive to use. Intrinsically, once enough people start using it. So their idea was, let's make transactions of Bitcoin back and forth, not just back and forth between two people, but between anyone in the world and me, and have it not actually touch the blockchain until some later time. And we'll go into that later, how that works. So that was their idea, but to actually make it work was this incredibly complicated Byzantine structure. And part of what made it complicated was something called malleability. So malleability is just a fancy word for change, really. If something's changeable, then it's malleable. And the problem is that intrinsically at that time, transactions on Bitcoin's blockchain were in some sense malleable in that when you created them, you could then create another version of them that was still valid before it got committed. So I know I'm getting a bit technical there, but basically there was a problem with the Bitcoin's blockchain that made it too changeable. And a solution came out called SegWit, but SegWit got all mixed up with something called the block size war. So while in 2016 these guys were coming up with this clever new scheme and some other people were getting excited about this clever new scheme, even though it was pretty complicated, they were still excited because we could get these transactions off the blockchain. A lot of other people were still arguing about, no, no, no, let's make the block sizes bigger because that way we can get more and more transactions on the blockchain. Okay, so the argument was about should we go off-chain or should we make the blockchain bigger to support more and more transactions. So Jonathan Beer here wrote a book recently called the Block Size War, I think about last year sometime. And it's quite interesting, if you don't know the history of the Block Size War, it is kind of interesting, but that's not the topic of our talk.

Reviewer: ```suggestion
what we're gonna see in the next couple of years is in the background, something else was happening. So this is January 2016, roughly. We have these two guys, Tadge Dryja and Joseph Poon, writing a paper called The Lightning Network Paper. Initially it was something just in the back quarters because it was just technical people, a few developers here and there and so on, researchers who were even remotely interested in this. Because actually this paper was kind of strange. It was written in a kind of strange way. I don't know how many people... Hands up if you have actually read the Lightning Network paper. I mean, more than like a few lines. More than a few lines. A few lines doesn't count. All right? There's a few. Well, well done. You're obviously very masochistic people because that paper is not an easy read. And this illustrates, I think, perfectly. This die is one of the final diagrams. It just shows you how complicated the structure of what they were describing was. So it was very difficult to understand, and it was based around a couple of little hacks because actually Bitcoin's blockchain intrinsically didn't support it. Now before I run on, what on earth am I talking about, the Lightning Network paper? What is the Lightning Network, just for the people who don't know what it is? It's basically the idea that given everything I've said in the last 10 minutes, we need a system where we can transfer Bitcoin without actually putting it on the Bitcoin blockchain. Because the Bitcoin blockchain is intrinsically pretty expensive to use. Intrinsically, once enough people start using it. So their idea was, let's make transactions of Bitcoin back and forth, not just back and forth between two people, but between anyone in the world and me, and have it not actually touch the blockchain until some later time. And we'll go into that later, how that works. So that was their idea, but to actually make it work was this incredibly complicated Byzantine structure. And part of what made it complicated was something called malleability. So malleability is just a fancy word for change, really. If something's changeable, then it's malleable. And the problem is that intrinsically at that time, transactions on Bitcoin's blockchain were in some sense malleable in that when you created them, you could then create another version of them that was still valid before it got committed. So I know I'm getting a bit technical there, but basically there was a problem with the Bitcoin's blockchain that made it too changeable. And a solution came out called SegWit, but SegWit got all mixed up with something called the block size war. So while in 2016 these guys were coming up with this clever new scheme and some other people were getting excited about this clever new scheme, even though it was pretty complicated, they were still excited because we could get these transactions off the blockchain. A lot of other people were still arguing about, no, no, no, let's make the block sizes bigger because that way we can get more and more transactions on the blockchain. Okay, so the argument was about should we go off-chain or should we make the blockchain bigger to support more and more transactions. So Jonathan Beer here wrote a book recently called the Block Size War, I think about last year sometime. And it's quite interesting, if you don't know the history of the Block Size War, it is kind of interesting, but that's not the topic of our talk.
```

## The Block Size War

>  So as we reach the end of 2016, it turns out there's a bunch of people actually starting to develop this thing called the Lightning Network, at least as software. Nobody's using it yet because this SegWit thing hadn't yet activated and so it was kind of like not very realistic to actually build it. Plus, because It was complicated, as I just explained to you. It is definitely a complicated system. Actually building out this software, LND, see lightning with Blockstream, Async, built Eclair, I'll talk about these guys in a minute. Building that software took time. Even getting just up to a first basic version took some time. So you had these companies or projects spring up. There was Lightning Labs, started by Elizabeth Stark here, along with LouLou, and they built this LND software. And this software will be behind a lot of the wallets that we're using later today. Blockstream, who started in 2015 with a whole sort of sidechain business. Rusty Russell, who's originally a Linux developer, and Christian Decker, who did a PhD in Bitcoin, sort of famously the first Bitcoin PhD. They were two of the people, and there were some other people who worked on another version of Lightning called C Lightning, and as I said before, Asank worked on Eclair. So you had at least three, and arguably more, different versions of this Lightning software being developed in 2016. And it got to the point, so by the end of 2016, we had the whole block size war raging. And in the background, at the same time, we had these guys putting together a set of protocols and agreements, because they had to make all of this complicated software talk to each other, and it was not trivial, yeah? I remember this, I put here the scaling Bitcoin sign, because in Milan in 2016, there was one of a series of scaling Bitcoin conferences, and one of the things that happened there was that was the first meetup. All these guys met up in a room. I remember there was a whole set of like 13 or 14 people, developers, and they were there solidly for two days just hashing it out, and what did they come up with? They came up with what's called nowadays the bolts.

Reviewer: ```suggestion
 So as we reach the end of 2016, it turns out there's a bunch of people actually starting to develop this thing called the Lightning Network, at least as software. Nobody's using it yet because this SegWit thing hadn't yet activated and so it was kind of like not very realistic to actually build it. Plus, because It was complicated, as I just explained to you. It is definitely a complicated system. Actually building out this software, LND, see lightning with Blockstream, Acinq, built Eclair, I'll talk about these guys in a minute. Building that software took time. Even getting just up to a first basic version took some time. So you had these companies or projects spring up. There was Lightning Labs, started by Elizabeth Stark here, along with LouLou, and they built this LND software. And this software will be behind a lot of the wallets that we're using later today. Blockstream, who started in 2015 with a whole sort of sidechain business. Rusty Russell, who's originally a Linux developer, and Christian Decker, who did a PhD in Bitcoin, sort of famously the first Bitcoin PhD. They were two of the people, and there were some other people who worked on another version of Lightning called C Lightning, and as I said before, Asank worked on Eclair. So you had at least three, and arguably more, different versions of this Lightning software being developed in 2016. And it got to the point, so by the end of 2016, we had the whole block size war raging. And in the background, at the same time, we had these guys putting together a set of protocols and agreements, because they had to make all of this complicated software talk to each other, and it was not trivial, yeah? I remember this, I put here the scaling Bitcoin sign, because in Milan in 2016, there was one of a series of scaling Bitcoin conferences, and one of the things that happened there was that was the first meetup. All these guys met up in a room. I remember there was a whole set of like 13 or 14 people, developers, and they were there solidly for two days just hashing it out, and what did they come up with? They came up with what's called nowadays the bolts.
```

## The Bolts

> So the bolts are the standards. They're a bit like RFCs for those of you who are engineers or internet people. They're a bit like RFCs. Basically describing what the protocol should do, what the protocol must do, what every software must agree to in order for all these lightning off-chain payments to work. And the outcome of all that is, in 2017, the software moves forward, they have something that kind of works, and we get by the summer of 2017, this sort of historic moment, it's kind of historic, kind of silly in a way, which is at Room 77 in Berlin you have Laulu, one of the engineers I just mentioned, paying for a beer with Lightning to, this is Jorg Platzer, the guy who owns, owned Room 77, actually with Testnet Lightning. So he didn't actually use real Bitcoin. He used testnet There's actually kind of a story behind that because why did they use? Why did they use testnet? I wonder if I can ask the audience Why did they use testnet lightning in the middle of 2017 and not mainchain Bitcoin lightning? Because Segwit hadn't been adopted at that point. Because Segwit hadn't been adopted at that point, thank you, that's the correct answer. So the weird thing is on Litecoin, they actually activated this Segwit thing, this thing that fixes malleability, some months earlier. So they were even trying to do this stuff on Litecoin, as well as on, this is confusing, Litecoin Lightning. They were doing it on Litecoin and on Bitcoin testnet, but they weren't doing it on main chain Bitcoin, because 2017 was like the end of the block size war, it was this massive explosion. I can't even begin to describe 2017 to you, It was the most crazy year ever from a whole bunch of different angles. But from one angle was we finally activated SegWit at the end of August, somebody correct me, or the beginning of August 2017, middle of August, Michael, anyone? Roughly, whatever, August 2017. So we activated SegWit in the midst of the hundred other things happening, which I'm not going to describe. But it meant that from the end of 2017, we could finally use this off-chain protocol called Lightning on the Bitcoin mainnet. Now, It didn't mean that everyone thought it was a great idea, but a lot of people did and it went forward. 

Reviewer: ```suggestion
So the bolts are the standards. They're a bit like RFCs for those of you who are engineers or internet people. They're a bit like RFCs. Basically describing what the protocol should do, what the protocol must do, what every software must agree to in order for all these lightning off-chain payments to work. And the outcome of all that is, in 2017, the software moves forward, they have something that kind of works, and we get by the summer of 2017, this sort of historic moment, it's kind of historic, kind of silly in a way, which is at Room 77 in Berlin you have Laolu, one of the engineers I just mentioned, paying for a beer with Lightning to, this is Jorg Platzer, the guy who owns, owned Room 77, actually with Testnet Lightning. So he didn't actually use real Bitcoin. He used testnet There's actually kind of a story behind that because why did they use? Why did they use testnet? I wonder if I can ask the audience Why did they use testnet lightning in the middle of 2017 and not mainchain Bitcoin lightning? Because Segwit hadn't been adopted at that point. Because Segwit hadn't been adopted at that point, thank you, that's the correct answer. So the weird thing is on Litecoin, they actually activated this Segwit thing, this thing that fixes malleability, some months earlier. So they were even trying to do this stuff on Litecoin, as well as on, this is confusing, Litecoin Lightning. They were doing it on Litecoin and on Bitcoin testnet, but they weren't doing it on main chain Bitcoin, because 2017 was like the end of the block size war, it was this massive explosion. I can't even begin to describe 2017 to you, It was the most crazy year ever from a whole bunch of different angles. But from one angle was we finally activated SegWit at the end of August, somebody correct me, or the beginning of August 2017, middle of August, Michael, anyone? Roughly, whatever, August 2017. So we activated SegWit in the midst of the hundred other things happening, which I'm not going to describe. But it meant that from the end of 2017, we could finally use this off-chain protocol called Lightning on the Bitcoin mainnet. Now, It didn't mean that everyone thought it was a great idea, but a lot of people did and it went forward. 
```

> ## Problems

Reviewer: Header breaking up a sentence

## Lightning wallets

 Some wallets, I've shown this one as an example of what happens when you actually try and work with Lightning at a sort of deeper level. If you run a full Lightning node and maybe have it connected to your home Raspberry Pi and you have various other models, servers and what have you, you can start getting involved in managing channels and oh, let's try and find liquidity, let's try and provide liquidity and get money for providing liquidity. So there's a whole rabbit hole of how deep you can go with Lightning. But for now, let's just think about using it just for sending and receiving payments. So briefly, I could mention my experience in El Salvador recently. I'm not sure if anyone, yeah, maybe one or two people were in El Salvador with me, so they know about it. 

## Summary

> How can I summarize this quickly? Because I'm kind of running on a little bit too long. So you've got the Bitcoin, it's very much a tale of two cities, the El Salvador experience. You've got the Bitcoin beach, famous project, in a place called El Zonte, literally on the beach, of course. Very small place, very, we could say underdeveloped. Few restaurants, few hotels, and basically everyone there has, like as a project, started using Bitcoin and also Lightning for payments for various things. And they tend to use something called the Bitcoin Beach Wallet, which is a really strange hybrid. It's not really like the things I just showed you, which you just install on your phone. It's a centralized custodial service by a company called Galloy. And when you scan their QR codes, maybe later one of you might want to try that one particularly, you'll see that you actually get sent to a webpage, not to a Lightning wallet. You have to go to a webpage, and the webpage then hosts the Lightning payment invoice for this person, even though again, it's custodial. They themselves are not holding the keys. So it's a good example of how people make trade-offs in the real world. Is that just for their business when it comes to tax or is that help in accounting? I think it's a complicated question as to why they've gone down that road, but I think it's mostly about the technological difficulty of it makes it a lot simpler for a person who doesn't have a detailed technical background or the time to get involved in nodes and whatnot. I think it's an interesting, I wanna move on, but I think it's a good question. Maybe we can talk about it in the second hour, yeah? Chivo, I wanna mention about Chivo, it's very important. Chivo, this is an example of the better aspect of Chivo, which their ATMs actually successfully, like will give you dollars in return for Bitcoin. But it's main chain only on the ATMs. Now their mobile wallet is quite frankly a complete disaster. I mean, by all accounts and by my own experience, a complete disaster. It does not work for Lightning anyway. It has by default a US dollar interface. By the way, guys, you were there. Correct me if I'm wrong, yeah? It has a US dollar interface by default. You can choose Bitcoin, but that will default to main chain Bitcoin. And then if you look really carefully, it will go to Lightning, yes. I was just gonna, you didn't finish, let me finish. The third option, if you look carefully, there is a button to choose Lightning, right? And perhaps you're gonna correct me in saying this, but from what I heard from other people and from myself, it tended to be the case that even if you could get a Lightning payment to work, that it wouldn't show up on the app. Did you find it differently? Yeah. It works for me all the time. It works for you every time, wow. So, okay. I was thinking about the merchant version. Yes. Personal version. And maybe the merchant one didn't work. The personal one worked. Right, that sounds right. Because I just remembered now, I did pay one person, it was like a barista, a coffee place, I paid his personal Chivo wallet, fine, you're exactly right. But the merchant one, I tried it a couple of times and I heard several other people saying that when you're in a restaurant and you try to pay them, your wallet would show that you'd paid $10 and their wallet would show nothing. So it was a disaster. I had to pay twice. But there have been several other problems in the early days of Chivo, and from what I've heard from people, it's really bad. And I'm kind of upset about it because I think it's kind of damaged the ordinary everyday Salvadorian's perspective of what Bitcoin adoption might be. 

Reviewer: ```suggestion
How can I summarize this quickly? Because I'm kind of running on a little bit too long. So you've got the Bitcoin, it's very much a tale of two cities, the El Salvador experience. You've got the Bitcoin beach, famous project, in a place called El Zonte, literally on the beach, of course. Very small place, very, we could say underdeveloped. Few restaurants, few hotels, and basically everyone there has, like as a project, started using Bitcoin and also Lightning for payments for various things. And they tend to use something called the Bitcoin Beach Wallet, which is a really strange hybrid. It's not really like the things I just showed you, which you just install on your phone. It's a centralized custodial service by a company called Galoy. And when you scan their QR codes, maybe later one of you might want to try that one particularly, you'll see that you actually get sent to a webpage, not to a Lightning wallet. You have to go to a webpage, and the webpage then hosts the Lightning payment invoice for this person, even though again, it's custodial. They themselves are not holding the keys. So it's a good example of how people make trade-offs in the real world. Is that just for their business when it comes to tax or is that help in accounting? I think it's a complicated question as to why they've gone down that road, but I think it's mostly about the technological difficulty of it makes it a lot simpler for a person who doesn't have a detailed technical background or the time to get involved in nodes and whatnot. I think it's an interesting, I wanna move on, but I think it's a good question. Maybe we can talk about it in the second hour, yeah? Chivo, I wanna mention about Chivo, it's very important. Chivo, this is an example of the better aspect of Chivo, which their ATMs actually successfully, like will give you dollars in return for Bitcoin. But it's main chain only on the ATMs. Now their mobile wallet is quite frankly a complete disaster. I mean, by all accounts and by my own experience, a complete disaster. It does not work for Lightning anyway. It has by default a US dollar interface. By the way, guys, you were there. Correct me if I'm wrong, yeah? It has a US dollar interface by default. You can choose Bitcoin, but that will default to main chain Bitcoin. And then if you look really carefully, it will go to Lightning, yes. I was just gonna, you didn't finish, let me finish. The third option, if you look carefully, there is a button to choose Lightning, right? And perhaps you're gonna correct me in saying this, but from what I heard from other people and from myself, it tended to be the case that even if you could get a Lightning payment to work, that it wouldn't show up on the app. Did you find it differently? Yeah. It works for me all the time. It works for you every time, wow. So, okay. I was thinking about the merchant version. Yes. Personal version. And maybe the merchant one didn't work. The personal one worked. Right, that sounds right. Because I just remembered now, I did pay one person, it was like a barista, a coffee place, I paid his personal Chivo wallet, fine, you're exactly right. But the merchant one, I tried it a couple of times and I heard several other people saying that when you're in a restaurant and you try to pay them, your wallet would show that you'd paid $10 and their wallet would show nothing. So it was a disaster. I had to pay twice. But there have been several other problems in the early days of Chivo, and from what I've heard from people, it's really bad. And I'm kind of upset about it because I think it's kind of damaged the ordinary everyday Salvadorian's perspective of what Bitcoin adoption might be. 
```

## Other things

> I'm not going to mention some of these other things. This is the Starbucks point of sale, which in my experience worked flawlessly every time. I think it's Ibex, does anyone know? I don't really care, it's some company that did it. It's kind of funny though, they show three different denominations. They throw dollars and then they show SATs, S-A-T, capital, and then they show BTC as well, just in case you get confused. It's just supposed to help you understand what the price is, I guess. But anyway, it works absolutely fine with more than one Lightning Wallet for me. Also McDonald's, which I'll show an example in a minute. Just to make pictures, like that's Bitcoin, this is my slideshow now, my holiday, I mean, it's a bit, sorry. Bitcoin Beach, it looks really nice in the sunset, cool. This is Hope House in El Zonte where you have the, some guys drove from Argentina or something, I don't even know the story here. This is something I just donated to, And that was the new mural by Bit Refill. It's a very interesting place to have a look. But San Salvador was different because they didn't have that Bitcoin beach wallet thing. So a lot of people were using Chivo and didn't really work very well with merchants. But On the other hand, McDonald's and Starbucks and a few chain restaurants were working fine. So it was all very strange. And you talk to the taxi drivers, half of them think it's great, half of them think it's terrible. I don't know what that proves. Yeah, so video time. So I just thought it might be amusing to finish this off by showing you what it's like to use it. So, if my eyesight can actually find the... There it is. So, this was a completely serendipitous meeting on the street. It's already scanned the QR code. All we need is a hotspot and 815. You got it? Woo! All right, all right, there we go. Every time I saw it, everybody's like, woo! Strict deals, coffee. I keep on talking about coffee, tea. So this very short video is basically, I met a guy called Tomer on the street. He's kind of a local here, actually. He's a journalist and videos and stuff. Him and his cameraman guy were talking to this guy who runs a coffee plantation, or owns a coffee plantation, I don't know. Anyway, he was getting interested in Bitcoin because he was showing them around and they said they gave him, I think they gave him, I know, they gave him Moon Wallet, he installed it. And I just met them on the street and I said, oh, you're selling coffee, I'll buy coffee with Bitcoin from you. So, it's a meme, right, You have to buy coffee with Bitcoin. You've got it, right? So he goes into the building, comes back out with a bag of coffee, which you can't quite see, but I'm holding it in my hand here. And he says, $10. And I paid him and it worked immediately. Now, I think that's a really cool example of how it can work. This is how it could be. It could be the case we could reach a point where everyone can reliably pay each other. $10.01 I paid for a $10 bag of coffee. I didn't know him, he didn't know me. It worked perfectly in literally like two or three seconds. Of course, it's not always going to be great like that, But that was an example where it was really good. Another example, it is a few seconds longer, but also a short video, is this, I actually just nabbed this from Twitter. This is, a lot of people here will know him. We're getting a bit of order at the company. We're a couple of minutes late. So it's what you'd expect, it's the normal McDonald's... That's a bit... Squeeze... That one's squeezed... Okay, that's water. Right, paint. And then that seems to shut down. And here... That's a bit of a break-up, we know that's what's going on. And with Bitcoin, see that? I'm going to get the seats jacked out. I'm going to put it here. Because I'm going to break up. We know that's not true. So with Bitcoin, see that? Yeah, we're going to see that. Yeah, nice one, I'm there. And I'm going to use my Allen bits wallet. Guys, like, thanks for the tips, but this is not the Allen bits tip jar. I think this is a good excuse to make a payment. I'm not a tip jar, I mean it's not bad to make a payment from a tip jar. There we are. Third big coin. I wonder what would be a coffee from McDonald's using like... There you go. Seven points. Scare it. Don't pay it. Just please. I want you to copy a copy from McDonald's using lightning. I know. All right, here we go. Is that 10 points? Ha, ha, ha, ha, ha, ha, ha, ha. Scan it. Double-click it. It's really slow. Come on, lightning. You can do it. LN bits will let. No, I mean it's just a little bit slow, but it works fine. Oh wow, look. Yeah, it does. Really? It works just fine, yeah. The paper works through. Yeah. Wow. Nice. Nice. Thanks for the company. And thanks for letting me pay for it. I'll get the LN bits, Chuck. So, I mean, that was a bit slower, but in my experience, I used the same thing several times and it was absolutely fine. Okay, well, I think I'm more or less done in the first half, so we can, if anybody has any questions or thoughts, please Ask, otherwise, oops, there we go. Otherwise we can move on to actually using it. Yeah, your turn. So this is your turn. So, yeah, questions. Between the nodes you can add liquidity, what did you mean by that? 

Reviewer: ```suggestion
I'm not going to mention some of these other things. This is the Starbucks point of sale, which in my experience worked flawlessly every time. I think it's Ibex, does anyone know? I don't really care, it's some company that did it. It's kind of funny though, they show three different denominations. They throw dollars and then they show SATs, S-A-T, capital, and then they show BTC as well, just in case you get confused. It's just supposed to help you understand what the price is, I guess. But anyway, it works absolutely fine with more than one Lightning Wallet for me. Also McDonald's, which I'll show an example in a minute. Just to make pictures, like that's Bitcoin, this is my slideshow now, my holiday, I mean, it's a bit, sorry. Bitcoin Beach, it looks really nice in the sunset, cool. This is Hope House in El Zonte where you have the, some guys drove from Argentina or something, I don't even know the story here. This is something I just donated to, And that was the new mural by Bitrefill. It's a very interesting place to have a look. But San Salvador was different because they didn't have that Bitcoin beach wallet thing. So a lot of people were using Chivo and didn't really work very well with merchants. But On the other hand, McDonald's and Starbucks and a few chain restaurants were working fine. So it was all very strange. And you talk to the taxi drivers, half of them think it's great, half of them think it's terrible. I don't know what that proves. Yeah, so video time. So I just thought it might be amusing to finish this off by showing you what it's like to use it. So, if my eyesight can actually find the... There it is. So, this was a completely serendipitous meeting on the street. It's already scanned the QR code. All we need is a hotspot and 815. You got it? Woo! All right, all right, there we go. Every time I saw it, everybody's like, woo! Strict deals, coffee. I keep on talking about coffee, tea. So this very short video is basically, I met a guy called Tomer on the street. He's kind of a local here, actually. He's a journalist and videos and stuff. Him and his cameraman guy were talking to this guy who runs a coffee plantation, or owns a coffee plantation, I don't know. Anyway, he was getting interested in Bitcoin because he was showing them around and they said they gave him, I think they gave him, I know, they gave him Moon Wallet, he installed it. And I just met them on the street and I said, oh, you're selling coffee, I'll buy coffee with Bitcoin from you. So, it's a meme, right, You have to buy coffee with Bitcoin. You've got it, right? So he goes into the building, comes back out with a bag of coffee, which you can't quite see, but I'm holding it in my hand here. And he says, $10. And I paid him and it worked immediately. Now, I think that's a really cool example of how it can work. This is how it could be. It could be the case we could reach a point where everyone can reliably pay each other. $10.01 I paid for a $10 bag of coffee. I didn't know him, he didn't know me. It worked perfectly in literally like two or three seconds. Of course, it's not always going to be great like that, But that was an example where it was really good. Another example, it is a few seconds longer, but also a short video, is this, I actually just nabbed this from Twitter. This is, a lot of people here will know him. We're getting a bit of order at the company. We're a couple of minutes late. So it's what you'd expect, it's the normal McDonald's... That's a bit... Squeeze... That one's squeezed... Okay, that's water. Right, paint. And then that seems to shut down. And here... That's a bit of a break-up, we know that's what's going on. And with Bitcoin, see that? I'm going to get the seats jacked out. I'm going to put it here. Because I'm going to break up. We know that's not true. So with Bitcoin, see that? Yeah, we're going to see that. Yeah, nice one, I'm there. And I'm going to use my Allen bits wallet. Guys, like, thanks for the tips, but this is not the Allen bits tip jar. I think this is a good excuse to make a payment. I'm not a tip jar, I mean it's not bad to make a payment from a tip jar. There we are. Third big coin. I wonder what would be a coffee from McDonald's using like... There you go. Seven points. Scare it. Don't pay it. Just please. I want you to copy a copy from McDonald's using lightning. I know. All right, here we go. Is that 10 points? Ha ha. Scan it. Double-click it. It's really slow. Come on, lightning. You can do it. LN bits will let. No, I mean it's just a little bit slow, but it works fine. Oh wow, look. Yeah, it does. Really? It works just fine, yeah. The paper works through. Yeah. Wow. Nice. Nice. Thanks for the company. And thanks for letting me pay for it. I'll get the LN bits, Chuck. So, I mean, that was a bit slower, but in my experience, I used the same thing several times and it was absolutely fine. Okay, well, I think I'm more or less done in the first half, so we can, if anybody has any questions or thoughts, please Ask, otherwise, oops, there we go. Otherwise we can move on to actually using it. Yeah, your turn. So this is your turn. So, yeah, questions. Between the nodes you can add liquidity, what did you mean by that? 
```
